src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    const COMPLEATLY_HEALTHY = 100;
+
+    this.health = COMPLEATLY_HEALTHY;
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(health, name) {
+    super(health, name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(health, name);
+  }
+
+  bite(prey) {
+    const BITE_OF_ANIMAL = 50;
+    const NOT_HEALTHY = 0;
```

Comment:
Same here. Move all variables to the top

Reply:
Thanks for your comment) I hope I understood correctly the comments)